### PR TITLE
feat: arb bridge swapper warnings

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Adds some warning/acknowledgements for Arbitrum Bridge deposits/withdraws in swapper.

See https://www.figma.com/design/IAtOWIGhClYAj7dxC4CzhJ/ARB-Bridge?node-id=75-3202&t=bSyZqQ7JyIQ0Revf-0

- [x] deposit warning
- [x] withdraw warning with new snap-like ack

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- contributes to https://github.com/shapeshift/web/issues/6979

> What protocols, transaction types or contract interactions might be affected by this PR?

Low - ensure existing ack are still happy visually

## Testing

- Try to do an Ethereum -> Arbitrum swap using the Arbitrum bridge and ensure the warning is displayed for it (see screenshots)
- Try to do an Arb -> Ethereum swap using the Arbitrum bridge and ensure the ack is displayed for it (see Jam)

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Deposit Warning

image

- Withdraw warning

image
image
image

### CODE REVIEW

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 1]
```diff
@@ -463,6 +475,22 @@ export const TradeInput = ({ isCompact }: TradeInputProps) => {
 
   const walletSupportsBuyAssetChain = useWalletSupportsChain(buyAsset.chainId, wallet)
 
+  const nativeAssetBridgeWarning: string | [string, InterpolationOptions] | undefined =
+    useMemo(() => {
+      if (!buyAssetFeeAsset) return
+      // TODO(gomes): Bring me in for all bridges?
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:481]: Not part of the spec but might as well at some point? Might require some more work to ensure we detect bridges properly without false positives however.

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 2]
```diff
@@ -463,6 +475,22 @@ export const TradeInput = ({ isCompact }: TradeInputProps) => {
...
+      const isBridge = (activeQuote as ArbitrumBridgeTradeQuote)?.direction === 'deposit'
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:482]: Only Arb deposits should show this warning for now

Comment (NeOMakinG) [reply src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:482]: Then we might want to call it `isArbitrumBridge`

Also the name is a lie for now, it should be either `isBridgeDeposit` or `isArbitrumBridgeDeposit`

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 3]
```diff
@@ -556,15 +591,19 @@ export const TradeInput = ({ isCompact }: TradeInputProps) => {
...
+    if ((activeQuote as ArbitrumBridgeTradeQuote)?.direction === 'withdrawal')
+      return setShouldShowArbitrumBridgeAcknowledgement(true)
     handleFormSubmit()
-  }, [handleFormSubmit, activeQuote?.isStreaming])
+  }, [activeQuote, handleFormSubmit])
 
   const handleTradeQuoteConfirm = useCallback(() => {
     if (isUnsafeQuote) return setShouldShowWarningAcknowledgement(true)
     if (activeQuote?.isStreaming) return setShouldShowStreamingAcknowledgement(true)
+    if ((activeQuote as ArbitrumBridgeTradeQuote)?.direction === 'withdrawal')
```
Comment (NeOMakinG) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:602]: Maybe we would prefer to use `isArbitrumBridgeTradeQuote(...)` than casting?

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx [hunk 4]
```diff
@@ -556,15 +591,19 @@ export const TradeInput = ({ isCompact }: TradeInputProps) => {
   // If the warning acknowledgement is shown, we need to handle the submit differently because we might want to show the streaming acknowledgement
   const handleWarningAcknowledgementSubmit = useCallback(() => {
     if (activeQuote?.isStreaming) return setShouldShowStreamingAcknowledgement(true)
+    if ((activeQuote as ArbitrumBridgeTradeQuote)?.direction === 'withdrawal')
```
Comment (NeOMakinG) [inline src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx:594]: same about `isArbitrumBridgeTradeQuote(...)`

File: src/components/Acknowledgement/Acknowledgement.tsx [hunk 5]
```diff
@@ -224,3 +233,69 @@ export const StreamingAcknowledgement = ({
...
+  const handleAcknowledge = useMemo(() => {
+    if (isDisabled) return
+
+    return props.onAcknowledge
+  }, [isDisabled, props])
+
+  const message = useMemo(
+    () => (
+      <>
+        <RawText>{translate('bridge.arbitrum.waitWarning')}</RawText>
```
Comment (NeOMakinG) [inline src/components/Acknowledgement/Acknowledgement.tsx:277]:
```suggestion
        <RawText as='span'>{translate('bridge.arbitrum.waitWarning')}</RawText>
```

or we will add another error:
image

**Review by NeOMakinG — APPROVED** [review]
Does what it says at runtime

Acknowledgement: https://jam.dev/c/15796d3d-986c-4f54-b133-833b0a03016d
Warning: 
image

I added a few comment, one is preferable blocking as it would generate an error but it's an easy to go, the other ones are nitpick namings

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx]: The idea is to eventually have all bridges show this warning, but agree in the meantime let's probably keep things declarative!
271401cf3

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx (hunk 4 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx]: 8fb16906b9

File: src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx (hunk 3 above)
Comment (gomesalexandre) [reply src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx]: Absolutely! 8fb16906b9

File: src/components/Acknowledgement/Acknowledgement.tsx (hunk 5 above)
Comment (gomesalexandre) [reply src/components/Acknowledgement/Acknowledgement.tsx]: 2abb277125259cb89028218160dbe4e7e105cde5

**Pull request merged**